### PR TITLE
Tech Writer Review of PPL Topic

### PR DESCRIPTION
**Pull request by WillCawley**
Tech Writer Suggestions

- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-custom-device-handbook/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Includes edits to the PPL .md file of the custom device developer guide.

### CODE REVIEW

File: docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md [hunk 1]
```diff
@@ -1,128 +1,183 @@
...
+##### Information Settings
 
-Select *Information* from the left menu and rename the PPL in the *Build Specification Name* field. Following convention, there are some rules to consider in order to organize the built files properly, so that, they can be easily copied/moved to the VeriStand Custom Device's directory once the build is complete (It would be worth mentioning that the *Post Build Action VI* mentioned in the next section, operates based on the same naming and path convention).
+In the Configuration Release PPL Properties dialog box, click **Information** to enter a new **Build Specification Name**.
+
+There are rules to consider when organizing the built files. These rules ensure that the files can be copied or moved to the VeriStand Custom Device's directory once built.
+
+**Note:** The Post Build Action VI operates based on the same naming and path convention.
 
  a) Custom Device System Explorer
```
Comment (WillCawley) [inline docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md:101]: @phutanu I am not sure I understood the point of these sub-divisions.

File: docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md [hunk 2]
```diff
@@ -1,128 +1,183 @@
...
+A *packed project library* (PPL) is a compiled LabVIEW library (*.lvlib*) that contains the VIs of that library. PPLs allow you to call public VIs similarly to their use in the original project library.
 
-Using PPLs for packaging VeriStand Custom Devices does not eliminate the usage of LLBs for the same purpose. A developer does not have to decide between using one or another. The two options can coexist as part of the same project. A developer can keep the existing LLB build specifications and simply add to the project the additional PPL ones. Of course, it might be necessary to modify the CD source code in order to be able to support both options. The XML configuration file would have to be modified as well to reflect the new loading paths for the CD elements.
+PPLs were first introduced in LabVIEW 2010. From LabVIEW 2017 onward, you can compile a PPL for newer versions of LabVIEW. This option is enabled by default.
 
-### Benefits for using PPL based Custom Devices
+**Note:** To change this setting, in the LabVIEW sidebar click **Advanced** and disable **Allow future versions of LabVIEW**.
```
Comment (phutanu) [inline docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md:7]:
```suggestion
**Note:** To change this setting, in the Packed Library build specification, click **Advanced** and disable **Allow future versions of LabVIEW to load this packed library**.
```

File: docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md [hunk 3]
```diff
@@ -1,128 +1,199 @@
...
+
+* The destination path should use the following structure: ``"..\built\Custom Device Name\Operating System``
+* The PLL file name should use the following format: [Custom Device Name] + [Engine] + [Operating System Name]
+
+The following image displays an example name and path of a Linux engine file.
 
 ![](images/Engine_Path.PNG)
 
-**Note:** Moving forward, we are supporting only one type of RT OS. This means there will be a total of two targets: Windows and Linux x64.
+**Note:** Future releases will only support one type of RT OS. There will be two targets: Windows and Linux x64.
```
Comment (phutanu) [inline docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md:137]:
```suggestion
**Note:** Starting with VeriStand 2021 only one type of real-time operating system is supported: Linux x64.
```

File: docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md [hunk 4]
```diff
@@ -1,128 +1,199 @@
...
 
-The best approach is to set such a VI as a *Post-Build Action* for your engine or configuration build specification (see below). You can also see it implemented in [FPGA Add-on Custom Device](https://github.com/ni/niveristand-fpga-addon-custom-device) and will be available in the upcoming version of the LabVIEW Custom Device Wizard.
+You can set this VI in LabVIEW as a Post-Build Action for your engine or configuration build specification.
+
+To set the action in LabVIEW, open the Engine Release PPL Properties dialog box and click **Pre/Post Build Actions** to enable **Excute VI after build**.
 
 ![](images/Execute_XML_updater.PNG)
 
-5. Build Files
+For an example of such a VI being implemented, refer to the [FPGA Add-on Custom Device](https://github.com/ni/niveristand-fpga-addon-custom-device). this custom device will be available in the upcoming version of the LabVIEW Custom Device Wizard.
```
Comment (phutanu) [inline docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md:191]:
```suggestion
For an example of such a VI being implemented, refer to the [FPGA Add-on Custom Device](https://github.com/ni/niveristand-fpga-addon-custom-device). this custom device will be available in the upcoming version of the NI VeriStand Custom Device Wizard.
```

File: docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md [hunk 5]
```diff
@@ -1,128 +1,199 @@
...
+Using a PPL to package a VeriStand custom device does not block the use of an LLB. The two options can coexist as a part of the same project. To use both packaging methods, retain the existing LLB build specifications and add the additional PPL build specifications to the project.
 
-* Since PPLs preserve the file hierarchy of the source project library, that solves the issue of internal naming conflicts, which is specific to LLBs (which causes compile-time warnings and extra files to be generated). Multiple LabVIEW project libraries can be included in the same PPL, even if those contain items with the same name. This is the same reason why LabVIEW classes could now be easier used when building VeriStand Custom Devices.
+**Note:** You may need to modify the custom device source code to support both options. You may also need to modify the XML configuration file to reflect the new loading paths for the custom device elements.
```
[the remark this answers is not in the document]
Comment (WillCawley) [reply docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md:11]: @phutanu How does this rewrite sound?
"To support PPL and LLB packaging, you may need to make the following modifications to your custom device: 
•	Update the custom device source code. 
•	Update the XML configuration file to reflect the new loading paths for the custom device elements.
•	Include mutation code for backwards compatibility of custom data types."

**Review by phutanu — APPROVED** [review]

**Comment by WillCawley** [comment]
@phutanu Are there any other reviewers I should add or is this ready to merge?

**Pull request merged**